src/Sharpliner/AzureDevOps/ConditionedExpressions/ConditionBuilder.cs

```diff
@@ -129,13 +200,40 @@ public Condition<T> And(params string[] expressions)
     public Condition<T> Or(params string[] expressions)
         => Link(new OrCondition<T>(expressions));
 
-    public Condition<T> StartsWith(string needle, string haystack)
+    public Condition<T> StartsWith(StringRuntimeExpression needle, StringRuntimeExpression haystack)
+        => new StartsWithCondition<T>(needle, haystack);
+
+    public Condition<T> StartsWith(IRuntimeExpression needle, StringRuntimeExpression haystack)
+        => new StartsWithCondition<T>(needle, haystack);
+
+    public Condition<T> StartsWith(StringRuntimeExpression needle, IRuntimeExpression haystack)
+        => new StartsWithCondition<T>(needle, haystack);
+
+    public Condition<T> StartsWith(IRuntimeExpression needle, IRuntimeExpression haystack)
         => new StartsWithCondition<T>(needle, haystack);
 
-    public Condition<T> EndsWith(string needle, string haystack)
+    public Condition<T> EndsWith(StringRuntimeExpression needle, StringRuntimeExpression haystack)
+        => new EndsWithCondition<T>(needle, haystack);
+
+    public Condition<T> EndsWith(IRuntimeExpression needle, StringRuntimeExpression haystack)
+        => new EndsWithCondition<T>(needle, haystack);
+
+    public Condition<T> EndsWith(StringRuntimeExpression needle, IRuntimeExpression haystack)
         => new EndsWithCondition<T>(needle, haystack);
 
-    public Condition<T> Contains(string needle, string haystack)
+    public Condition<T> EndsWith(IRuntimeExpression needle, IRuntimeExpression haystack)
+        => new EndsWithCondition<T>(needle, haystack);
+
+    public Condition<T> Contains(StringRuntimeExpression needle, StringRuntimeExpression haystack)
+        => new ContainsCondition<T>(needle, haystack);
+
+    public Condition<T> Contains(IRuntimeExpression needle, StringRuntimeExpression haystack)
+        => new ContainsCondition<T>(needle, haystack);
+
+    public Condition<T> Contains(StringRuntimeExpression needle, IRuntimeExpression haystack)
+        => new ContainsCondition<T>(needle, haystack);
+
+    public Condition<T> Contains(IRuntimeExpression needle, IRuntimeExpression haystack)
         => new ContainsCondition<T>(needle, haystack);
 
     public Condition In(string needle, string haystack)
```

Comment:
So this is my fault, not yours, but we're missing some APIs here. Would you mind completing those too? We can also do it as part of another PR but maybe it makes sense?

Something like 
```csharp
    public Condition In(string needle, string haystack)
        => new InCondition(needle, haystack);

    public Condition In(string needle, params string[] haystack)
        => new InCondition(needle, haystack);

    public Condition NotIn(string needle, string haystack)
        => new NotInCondition(needle, haystack);

    public Condition NotIn(string needle, params string[] haystack)
        => new NotInCondition(needle, haystack);
```
(not sure these might not need changing to `IRuntimeExpression` then.

What should be possible is something like this:
```csharp
Variables =
{
    If.In(variables.Build.SourceBranch, "refs/heads/feature/", parameters["allowedTags"], variables["foo"], variables.Build.Reason)
        .Variable("feature", "on"),
}
```

Reply:
I had to add two extra methods to accept a params object[] array. This is so the caller can mix variables/params (IRuntimeExpression) and strings within the same array.

But this should be done now if you'd like to take a look. Added a test for your above scenario too.